### PR TITLE
Mes-2298-tell-me-questions

### PR DESCRIPTION
**Pull request by gittins7**
## Description
 - Updated vehicle checks cat B+E modal, selector, actions & reducer to include required logic for tell me questions
 - Update vehicle-checks-questions component to use a lodash generated id, instead of an index passed in via an [input] for each component instance
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [x] One review from each scrum team
- [x] Squashed commit contains the JIRA ticket number

## Screenshot

Browser -
Screenshot 2019-11-13 at 14 29 03

Simulator -
Screenshot 2019-11-13 at 15 28 43

### CODE REVIEW

**Comment by ammarhaiderbjss** [comment]
@gittins7 Looks good, but can we add a screenshot from either simulator / real device please?

File: src/pages/waiting-room-to-car/cat-be/components/vehicle-checks-modal/vehicle-checks-modal.cat-be.page.ts [hunk 1]
```diff
@@ -68,6 +76,13 @@ export class VehicleChecksCatBEModal {
         select(getVehicleChecksCatBe),
         select(getSelectedShowMeQuestions),
       ),
+      tellMeQuestions$: this.store$.pipe(
+        select(getTests),
```
Comment (szabikr) [inline src/pages/waiting-room-to-car/cat-be/components/vehicle-checks-modal/vehicle-checks-modal.cat-be.page.ts]: Would extract 
```
const currentTest$ = this.store$.pipe(
    select(getTests),
    select(getCurrentTest),
);
```

And reuse it when creating these observables

**Review by szabikr — COMMENTED** [review]
All looks good! A minor improvement I mentioned in the comments

File: src/pages/waiting-room-to-car/cat-be/components/vehicle-checks-modal/vehicle-checks-modal.cat-be.page.ts (hunk 1 above)
Comment (gittins7) [reply src/pages/waiting-room-to-car/cat-be/components/vehicle-checks-modal/vehicle-checks-modal.cat-be.page.ts]: ahh right ok, do you mean something like this?:-

```
  ngOnInit(): void {
    const currentTest$ = this.store$.pipe(
      select(getTests),
      select(getCurrentTest),
    )
    this.pageState = {
      candidateName$: currentTest$.pipe(
        select(getJournalData),
        select(getCandidate),
        select(getUntitledCandidateName),
      ),
      showMeQuestions$: currentTest$.pipe(
        select(getTestData),
        select(getVehicleChecksCatBe),
        select(getSelectedShowMeQuestions),
      ),
      tellMeQuestions$: currentTest$.pipe(
        select(getTestData),
        select(getVehicleChecksCatBe),
        select(getSelectedTellMeQuestions),
      )
    };
  }
```

Comment (szabikr) [reply src/pages/waiting-room-to-car/cat-be/components/vehicle-checks-modal/vehicle-checks-modal.cat-be.page.ts]: Yeah that's right

**Review by ammarhaiderbjss — APPROVED** [review]
😍

**Review by rossfauldsbjss — APPROVED** [review]
looks good

**Pull request merged**